kubernetes/deployments_kubeclient_blackbox_test.go

```diff
@@ -0,0 +1,452 @@
+}
+
+func (testMetrics) GetNetworkRecvMetrics(pods []*v1.Pod, namespace string, startTime time.Time) (*app.TimedNumberTuple, error) {
+	return nil, nil // TODO add fake impl when tests exercise this code
+}
+
+func (testMetrics) GetNetworkRecvMetricsRange(pods []*v1.Pod, namespace string, startTime time.Time, endTime time.Time,
+	limit int) ([]*app.TimedNumberTuple, error) {
+	return nil, nil // TODO add fake impl when tests exercise this code
+}
+
+func TestGetMetrics(t *testing.T) {
+	kubeGetter := &testKubeGetter{
+		cmInput: defaultConfigMapInput,
+	}
+	metricsGetter := &testMetricsGetter{}
+
+	token := "myToken"
+	testCases := []struct {
+		clusterURL    string
+		expectedURL   string
+		shouldSucceed bool
+	}{
+		{"https://api.myCluster.url:443/cluster", "https://metrics.myCluster.url", true},
+		{"https://myCluster.url:443/cluster", "", false},
+	}
+
+	for _, testCase := range testCases {
+		config := &kubernetes.KubeClientConfig{
+			ClusterURL:        testCase.clusterURL,
+			BearerToken:       token,
+			UserNamespace:     "myNamespace",
+			KubeRESTAPIGetter: kubeGetter,
+			MetricsGetter:     metricsGetter,
+		}
+		_, err := kubernetes.NewKubeClient(config)
+		if testCase.shouldSucceed {
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			assert.Equal(t, testCase.expectedURL, metricsGetter.config.MetricsURL, "Incorrect Metrics URL")
+			assert.Equal(t, token, metricsGetter.config.BearerToken, "Incorrect bearer token")
+		} else {
+			if err == nil {
+				t.Error("Expected error, but was successful")
+			}
+		}
+	}
+}
+
+func TestConfigMapEnvironments(t *testing.T) {
+	testCases := []*configMapInput{
+		{
+			labels: map[string]string{"provider": "fabric8"},
+			data: map[string]string{
+				"run":   "name: Run\nnamespace: my-run\norder: 1",
+				"stage": "name: Stage\nnamespace: my-stage\norder: 0",
+			},
+		},
+		{
+			labels: map[string]string{"provider": "fabric8"},
+			data:   map[string]string{},
+		},
+		{
+			labels: map[string]string{"provider": "fabric8"},
+			data: map[string]string{
+				"run": "name: Run\nnamespace my-run\norder: 1", // Missing colon
+			},
+			shouldFail: true,
+		},
+		{
+			labels: map[string]string{"provider": "fabric8"},
+			data: map[string]string{
+				"run": "name: Run\nns: my-run\norder: 1", // Missing namespace
+			},
+			shouldFail: true,
+		},
+		{
+			shouldFail: true, // No provider
+		},
+	}
+	kubeGetter := &testKubeGetter{}
+	metricsGetter := &testMetricsGetter{}
+	userNamespace := "myNamespace"
+	config := &kubernetes.KubeClientConfig{
+		ClusterURL:        "http://api.myCluster",
+		BearerToken:       "myToken",
+		UserNamespace:     userNamespace,
+		KubeRESTAPIGetter: kubeGetter,
+		MetricsGetter:     metricsGetter,
+	}
+
+	expectedName := "fabric8-environments"
+	for _, testCase := range testCases {
+		kubeGetter.cmInput = testCase
+		_, err := kubernetes.NewKubeClient(config)
+		if testCase.shouldFail {
+			assert.Error(t, err)
+		} else {
+			if !assert.NoError(t, err) {
```

Comment:
what are you trying to do here ?

Reply:
(changed to require and cleaned up) the list of test data has some cases that pass, and some that fail.  The code is now cleaner to relfect intent.

Reply:
ok, good :)